### ISSUE TITLE
Questions about the loss computation

### ISSUE THREAD

**Issue by tcourat** [ChenHongruixuan/ChangeMamba#106]
HI, I have a few questions about the loss computations that do not seem intuitive to me.

1. What is this semantic similarity loss you are using here : https://github.com/ChenHongruixuan/MambaCD/blob/96c3240f0ca5ffdbee682a70af03e1d7cccf70aa/changedetection/script/train_MambaSCD.py#L130
It is not mentioned in the paper. And it seems a bit weird to me as it tries to make the semantic logits from T1 and T2 the same only when the location does not change. Should we not also enforce that the logits need to be different when the location changes ? By doing something similar to
`sim_loss = (1 - mask)*diff + mask*(1 - diff)` where mask=0 if it does not change (otherwise 1) and diff is the MSE of semantic logits.

2. Why are you ignoring index 255 when computing the change detection loss https://github.com/ChenHongruixuan/MambaCD/blob/96c3240f0ca5ffdbee682a70af03e1d7cccf70aa/changedetection/script/train_MambaSCD.py#L117
I can understand that you ignore the non-change location to compute the semantic losses, but how can the binary change loss works if you ignore entirely one class, e.g the class "no change" and only backpropagate on location where it changes ? Maybe I misunderstood.

3. Where do the wheighting comes from in your final total loss expression
https://github.com/ChenHongruixuan/MambaCD/blob/96c3240f0ca5ffdbee682a70af03e1d7cccf70aa/changedetection/script/train_MambaSCD.py#L133
Is it something purely experimental that worked better in your various train experiments ? Or is it motivated by something else ? 

Thanks

**Comment by ChenHongruixuan** [comment]
Hi @tcourat ,

thank you so much for your interest to our work and your question. 

1. The reason for adopting this loss function is that the SECOND dataset does not provide semantic category annotations for pixels in unchanged regions. However, such information can also be helpful for detection (an idea inspired by [our another paper](https://ieeexplore.ieee.org/abstract/document/10551264)). Therefore, we designed this loss function to explicitly encourage the model to produce the same classification results for unchanged regions. 
 
2.  In the change labels, 0 represents unchanged pixels and 1 represents changed pixels. Therefore, setting the ignore index to 255 has no effect. This line of code was likely used for private testing or copied from our another script.

3. The weights of the loss function are empirically set. Of course you are free to adjust them yourself — it might even lead to better performance.

Best,
